aiida/tools/archive/create.py

```diff
@@ -568,9 +568,10 @@ def _stream_repo_files(
 ) -> None:
     """Collect all repository object keys from the nodes, then stream the files to the archive."""
     keys = set(
-        orm.Node.collection(backend).iter_repo_keys(filters={'id': {
-            'in': list(node_ids)
-        }}, batch_size=batch_size)
+        orm.nodes.node.NodeCollection(orm.Node, backend
+                                      ).iter_repo_keys(filters={'id': {
+                                          'in': list(node_ids)
+                                      }}, batch_size=batch_size)
```

Comment:
Same here?
```suggestion
        orm.nodes.node.NodeCollection(backend=backend
                                      ).iter_repo_keys(filters={'id': {
                                          'in': list(node_ids)
                                      }}, batch_size=batch_size)
```

Reply:
yeh again, this won't work until your proposed change to `Collection.__init__` is implemented